warehouse/integrations/azure-synapse/azure-synapse.go

```diff
@@ -770,11 +770,11 @@ func (as *AzureSynapse) FetchSchema(warehouse model.Warehouse) (schema, unrecogn
 	return
 }
 
-func (as *AzureSynapse) LoadUserTables() map[string]error {
+func (as *AzureSynapse) LoadUserTables(context.Context) map[string]error {
 	return as.loadUserTables()
```

Comment:
Why not propagate the context to `as.loadUserTables` as well? It looks like [it can be benefit from it](https://github.com/rudderlabs/rudder-server/blob/fix.postgres-load-table/warehouse/integrations/azure-synapse/azure-synapse.go#L511):

```go
	// BEGIN TRANSACTION
	tx, err := as.DB.BeginTx(ctx)
```

Reply:
Same with `LoadTable` where for example we have code like the following that can be updated:

```go
fileNames, err := as.LoadFileDownLoader.Download(context.TODO(), tableName)
```